src/PlotlyPanel.tsx

```diff
@@ -144,6 +128,19 @@ export const PlotlyPanel: React.FC<Props> = (props) => {
     });
   };
 
+  const handlePlotRelayout = (event: Readonly<Plotly.PlotRelayoutEvent>) => {
+    const { "xaxis.range[0]": xAxisMin, "xaxis.range[1]": xAxisMax} = event;
+
+    if (typeof xAxisMin === 'string' && typeof xAxisMax === 'string') {
+      const from = dateTimeParse(xAxisMin);
+      const to = dateTimeParse(xAxisMax);
+
+      if (from.isValid() && to.isValid()) {
+        props.onChangeTimeRange({ from: from.valueOf(), to: to.valueOf() });
+      }
+    }
+  };
```

Comment:
This function handles updating the dashboard-wide time filter when users zoom/pan on a graph.